src/mscorlib/src/System/Environment.cs

```diff
@@ -701,20 +690,19 @@ private static IDictionary GetEnvironmentVariablesCore(EnvironmentVariableTarget
             }
             else
             {
-                throw new ArgumentException(SR.Format(SR.Arg_EnumIllegalVal, (int)target));
+                throw new ArgumentOutOfRangeException(nameof(target), target, SR.Format(SR.Arg_EnumIllegalVal, target));
```

Comment:
This matches what ValidateTargets() throws (and what the CoreFx tests expect.)

Others that are throwing ArgumentException for a bad Target are wrong too but they're unreachable since someone calls ValidateTargets() first. I'm keeping the changes minimal on this one since I expect this to be replaced through the shared partition eventually.